diff --git a/submodules/fb2k_utils/src/qwr/qwr_exception.h b/submodules/fb2k_utils/src/qwr/qwr_exception.h
index 7e75b04..de7a01c 100644
--- a/submodules/fb2k_utils/src/qwr/qwr_exception.h
+++ b/submodules/fb2k_utils/src/qwr/qwr_exception.h
@@ -12,7 +12,7 @@ class QwrException
 public:
     template <typename... Args>
     explicit QwrException( qwr::u8string_view errorMessage, Args&&... errorMessageFmtArgs )
-        : std::runtime_error( fmt::format( errorMessage, std::forward<Args>( errorMessageFmtArgs )... ) )
+        : std::runtime_error( fmt::format( fmt::runtime( errorMessage ), std::forward<Args>( errorMessageFmtArgs )... ) )
     {
     }

