diff --git a/submodules/fb2k_utils/src/qwr/error_popup.cpp b/submodules/fb2k_utils/src/qwr/error_popup.cpp
index 17ceb22..04d12b0 100644
--- a/submodules/fb2k_utils/src/qwr/error_popup.cpp
+++ b/submodules/fb2k_utils/src/qwr/error_popup.cpp
@@ -9,26 +9,32 @@ namespace qwr

 void ReportErrorWithPopup( const std::string& title, const std::string& errorText )
 {
-    const auto report = [title, errorText] {
-        assert( core_api::assert_main_thread() );
+    ::fb2k::inMainThread2( [title, errorText]
+    {
+        FB2K_console_formatter() << title << ":\n"  << errorText << "\n";

-        FB2K_console_formatter() << title << ":\n"
-                                 << errorText << "\n";
         qwr::DelayedExecutor::GetInstance().AddTask( [errorText, title] {
             popup_message::g_show( errorText.c_str(), title.c_str() );
         } );

         MessageBeep( MB_ICONASTERISK );
-    };
+    });
+}

-    if ( core_api::is_main_thread() )
-    {
-        report();
-    }
-    else
+void ReportError( const std::string& title, const std::string& errorText )
+{
+    ::fb2k::inMainThread2 ( [title, errorText]
     {
-        ::fb2k::inMainThread2( report );
-    }
+        FB2K_console_formatter() << title << ":\n"  << errorText << "\n";
+
+        qwr::DelayedExecutor::GetInstance()
+            .AddTask ( [errorText, title]
+            {
+                popup_message::g_show( errorText.c_str(), title.c_str() );
+            });
+
+        MessageBeep( MB_ICONASTERISK );
+    });
 }

 } // namespace qwr
diff --git a/submodules/fb2k_utils/src/qwr/error_popup.h b/submodules/fb2k_utils/src/qwr/error_popup.h
index 71baf87..549b2f2 100644
--- a/submodules/fb2k_utils/src/qwr/error_popup.h
+++ b/submodules/fb2k_utils/src/qwr/error_popup.h
@@ -6,5 +6,6 @@ namespace qwr
 {

 void ReportErrorWithPopup( const std::string& title, const std::string& errorText );
+void ReportError( const std::string& title, const std::string& errorText );

 } // namespace qwr
