diff --git a/submodules/lexilla/src/lexilla.mak b/submodules/lexilla/src/lexilla.mak
index b04551a0..8109b685 100644
--- a/submodules/lexilla/src/lexilla.mak
+++ b/submodules/lexilla/src/lexilla.mak
@@ -10,8 +10,8 @@
 
 .SUFFIXES: .cxx
 
-DIR_O=.
-DIR_BIN=..\bin
+DIR_O=$(QWR_TEMPDIR)
+DIR_BIN=$(QWR_BINDIR)
 
 LEXILLA=$(DIR_BIN)\lexilla.dll
 LIBLEXILLA=$(DIR_BIN)\liblexilla.lib
@@ -40,8 +40,12 @@ CXXFLAGS=-Zi -TP -MP -W4 -EHsc -std:c++17 $(CRTFLAGS)
 CXXDEBUG=-Od -MTd -DDEBUG
 CXXNDEBUG=-O2 -MT -DNDEBUG -GL
 NAME=-Fo
-LDFLAGS=-OPT:REF -LTCG -IGNORE:4197 -DEBUG $(SUBSYSTEM) $(CETCOMPAT)
+LDFLAGS=-OPT:REF -IGNORE:4197 -DEBUG $(SUBSYSTEM) $(CETCOMPAT)
 LDDEBUG=
+LDRELEASE=-LTCG
+LIBFLAGS=
+LIBDEBUG=
+LIBRELEASE=/LTCG
 LIBS=
 NOLOGO=-nologo
 
@@ -54,8 +58,11 @@ LDFLAGS=$(LDFLAGS) $(NOLOGO)
 !IFDEF DEBUG
 CXXFLAGS=$(CXXFLAGS) $(CXXDEBUG)
 LDFLAGS=$(LDDEBUG) $(LDFLAGS)
+LIBFLAGS=$(LIBDEBUG) $(LIBFLAGS)
 !ELSE
 CXXFLAGS=$(CXXFLAGS) $(CXXNDEBUG)
+LDFLAGS=$(LDRELEASE) $(LDFLAGS)
+LIBFLAGS=$(LIBRELEASE) $(LIBFLAGS)
 !ENDIF
 
 SCINTILLA_INCLUDE = ../../scintilla/include
@@ -63,7 +70,7 @@ SCINTILLA_INCLUDE = ../../scintilla/include
 INCLUDEDIRS=-I../include -I$(SCINTILLA_INCLUDE) -I../src -I../lexlib
 CXXFLAGS=$(CXXFLAGS) $(INCLUDEDIRS)
 
-all:	$(SCINTILLA_INCLUDE) $(LEXILLA) $(LIBLEXILLA)
+all: $(DIR_O) $(DIR_BIN) $(SCINTILLA_INCLUDE) $(LEXILLA) $(LIBLEXILLA)
 
 clean:
 	-del /q $(DIR_O)\*.obj $(DIR_O)\*.o $(DIR_O)\*.pdb \
@@ -73,6 +80,12 @@ clean:
 depend:
 	pyw DepGen.py
 
+$(DIR_O):
+    -@ if NOT EXIST $(DIR_O) mkdir $(DIR_O)
+
+$(DIR_BIN):
+    -@ if NOT EXIST $(DIR_BIN) mkdir $(DIR_BIN)
+
 $(SCINTILLA_INCLUDE):
 	@echo Scintilla must be installed at ../../scintilla to provide access to Scintilla headers.
 
@@ -107,7 +120,7 @@ $(LEXILLA): $(LEXILLA_OBJS) LexillaVersion.res
 	$(LD) $(LDFLAGS) -DEF:Lexilla.def -DLL -OUT:$@ $** $(LIBS)
 
 $(LIBLEXILLA): $(LEXILLA_OBJS)
-	LIB -OUT:$@ $**
+	LIB $(LIBFLAGS) -OUT:$@ $**
 
 # Define how to build all the objects and what they depend on
 
