diff a/scintilla b/scintilla
--- a/scintilla/win32/scintilla.mak
+++ b/scintilla/win32/scintilla.mak
@@ -26,10 +26,10 @@
 
 CRTFLAGS=-D_CRT_SECURE_CPP_OVERLOAD_STANDARD_NAMES=1 -D_CRT_SECURE_NO_DEPRECATE=1 -D_SCL_SECURE_NO_WARNINGS=1 $(XP_DEFINE)
 CXXFLAGS=-Zi -TP -MP -W4 -EHsc -std:c++latest $(CRTFLAGS)
-CXXDEBUG=-Od -MTd -DDEBUG
-CXXNDEBUG=-O1 -MT -DNDEBUG -GL
+CXXDEBUG=-Od -MDd -DDEBUG
+CXXNDEBUG=-O1 -MD -DNDEBUG -GL
 NAME=-Fo
-LDFLAGS=-OPT:REF -LTCG -IGNORE:4197 -DEBUG $(XP_LINK)
+LDFLAGS=-OPT:REF -IGNORE:4197 -DEBUG $(XP_LINK)
 LDDEBUG=
 LIBS=KERNEL32.lib USER32.lib GDI32.lib IMM32.lib OLE32.lib OLEAUT32.lib MSIMG32.lib
 NOLOGO=-nologo
--- a/scintilla/win32/ScintillaWin.cxx
+++ b/scintilla/win32/ScintillaWin.cxx
@@ -140,7 +140,7 @@
 
 namespace {
 
-const TCHAR callClassName[] = TEXT("CallTip");
+const TCHAR callClassName[] = TEXT("SMP_CallTip");
 
 void *PointerFromWindow(HWND hWnd) {
 	return reinterpret_cast<void *>(::GetWindowLongPtr(hWnd, 0));
@@ -3222,7 +3222,7 @@
 	wndclass.hCursor = NULL;
 	wndclass.hbrBackground = NULL;
 	wndclass.lpszMenuName = NULL;
-	wndclass.lpszClassName = L"Scintilla";
+	wndclass.lpszClassName = L"SMP_Scintilla";
 	wndclass.hIconSm = 0;
 	scintillaClassAtom = ::RegisterClassExW(&wndclass);
 	bool result = 0 != scintillaClassAtom;
