diff a/submodules/scintilla b/submodules/scintilla
index 2a7a839c..44bf53b2 100644
--- a/submodules/scintilla/lexilla/src/lexilla.mak
+++ b/submodules/scintilla/lexilla/src/lexilla.mak
@@ -37,8 +37,12 @@ CXXFLAGS=-Zi -TP -MP -W4 -EHsc -std:c++17 $(CRTFLAGS)
 CXXDEBUG=-Od -MTd -DDEBUG
 CXXNDEBUG=-O1 -MT -DNDEBUG -GL
 NAME=-Fo
-LDFLAGS=-OPT:REF -LTCG -IGNORE:4197 -DEBUG $(SUBSYSTEM)
+LDFLAGS=-OPT:REF -IGNORE:4197 -DEBUG $(SUBSYSTEM)
 LDDEBUG=
+LDRELEASE=-LTCG
+LIBFLAGS=
+LIBDEBUG=
+LIBRELEASE=/LTCG
 LIBS=
 NOLOGO=-nologo
 
@@ -51,9 +55,12 @@ LDFLAGS=$(LDFLAGS) $(NOLOGO)
 !IFDEF DEBUG
 CXXFLAGS=$(CXXFLAGS) $(CXXDEBUG)
 LDFLAGS=$(LDDEBUG) $(LDFLAGS)
+LIBFLAGS=$(LIBDEBUG) $(LIBFLAGS)
 !ELSE
 CXXFLAGS=$(CXXFLAGS) $(CXXNDEBUG)
-!ENDIF
+LDFLAGS=$(LDRELEASE) $(LDFLAGS)
+LIBFLAGS=$(LIBRELEASE) $(LIBFLAGS)
+!ENDIF
 
 INCLUDEDIRS=-I../../include -I../../src -I../../lexlib
 CXXFLAGS=$(CXXFLAGS) $(INCLUDEDIRS)
@@ -98,7 +105,7 @@ $(LEXILLA): $(LEXILLA_OBJS)
 	$(LD) $(LDFLAGS) -DEF:Lexilla.def -DLL -OUT:$@ $** $(LIBS)
 
 $(LIBLEXILLA): $(LEXILLA_OBJS)
-	LIB -OUT:$@ $**
+	LIB $(LIBFLAGS) -OUT:$@ $**
 
 # Define how to build all the objects and what they depend on
 
index 4a652f42..d51c77af 100644
--- a/submodules/scintilla/win32/scintilla.mak
+++ b/submodules/scintilla/win32/scintilla.mak
@@ -38,8 +38,12 @@ CXXFLAGS=-Zi -TP -MP -W4 -EHsc -std:c++17 $(CRTFLAGS)
 CXXDEBUG=-Od -MTd -DDEBUG
 CXXNDEBUG=-O1 -MT -DNDEBUG -GL
 NAME=-Fo
-LDFLAGS=-OPT:REF -LTCG -IGNORE:4197 -DEBUG $(SUBSYSTEM)
+LDFLAGS=-OPT:REF -IGNORE:4197 -DEBUG $(SUBSYSTEM)
 LDDEBUG=
+LDRELEASE=-LTCG
+LIBFLAGS=
+LIBDEBUG=
+LIBRELEASE=/LTCG
 LIBS=KERNEL32.lib USER32.lib GDI32.lib IMM32.lib OLE32.lib OLEAUT32.lib MSIMG32.lib
 NOLOGO=-nologo
 
@@ -56,8 +60,11 @@ CXXFLAGS=$(CXXFLAGS) -DNO_CXX11_REGEX
 !IFDEF DEBUG
 CXXFLAGS=$(CXXFLAGS) $(CXXDEBUG)
 LDFLAGS=$(LDDEBUG) $(LDFLAGS)
+LIBFLAGS=$(LIBDEBUG) $(LIBFLAGS)
 !ELSE
 CXXFLAGS=$(CXXFLAGS) $(CXXNDEBUG)
+LDFLAGS=$(LDRELEASE) $(LDFLAGS)
+LIBFLAGS=$(LIBRELEASE) $(LIBFLAGS)
 !ENDIF
 
 INCLUDES=-I../include -I../src -I../lexlib
@@ -163,7 +170,7 @@ $(LEXCOMPONENT): $(LEXCOMPONENT_OBJS) $(DIR_O)\ScintRes.res
 	$(LD) $(LDFLAGS) -DEF:Scintilla.def -DLL -OUT:$@ $** $(LIBS)
 
 $(LIBSCI): $(COMPONENT_OBJS)
-	LIB /OUT:$@ $**
+	LIB $(LIBFLAGS) /OUT:$@ $**
 
 # Define how to build all the objects and what they depend on
 
--- a/submodules/scintilla/win32/ScintillaWin.cxx
+++ b/submodules/scintilla/win32/ScintillaWin.cxx
@@ -150,7 +150,7 @@
 
 namespace {
 
-const TCHAR callClassName[] = TEXT("CallTip");
+const TCHAR callClassName[] = TEXT("SMP_CallTip");
 
 void SetWindowID(HWND hWnd, int identifier) noexcept {
 	::SetWindowLongPtr(hWnd, GWLP_ID, identifier);
@@ -3337,7 +3337,7 @@
 	wndclass.lpfnWndProc = ScintillaWin::SWndProc;
 	wndclass.cbWndExtra = sizeof(ScintillaWin *);
 	wndclass.hInstance = hInstance;
-	wndclass.lpszClassName = L"Scintilla";
+	wndclass.lpszClassName = L"SMP_Scintilla";
 	scintillaClassAtom = ::RegisterClassExW(&wndclass);
 	const bool result = 0 != scintillaClassAtom;
 
